diff.py

```diff
@@ -1411,7 +1435,7 @@ def dump_objfile(
     if not os.path.isfile(objfile):
         fail(f"Not able to find .o file for function: {objfile} is not a file.")
 
-    refobjfile = "expected/" + objfile
+    refobjfile = project.expected_directory + objfile
```

Comment:
```suggestion
    refobjfile = os.path.join(project.expected_directory, objfile)
```
(which will make it work even if a trailing / is missing)